### PR TITLE
Simplify and update support view of references

### PR DESCRIPTION
**Pull request by frankieroberto**
## Context

We’ve made a load of changes to the references process, but haven't updated the Support interface yet.

I've done a quick redesign of the component, mainly aiming to simplify it and re-use most of the design from how this appears in the Manage interface.

The code needs a review and might need refactoring, as I've done this quite quickly.

I've removed all the "Can [xxxx]?" rows for now, as I'm not sure that these are still needed or helpful?

This also fixes a longstanding bug where "Relationship confirmed by referee" says "Yes" even if they've not given a reference or answered that question yet.

Trello card: https://trello.com/c/iRpCec50/791-update-support-service-with-changes-to-references

### Before

Screenshot 2022-10-26 at 14 48 44

Screenshot 2022-10-26 at 14 49 20

### After

Screenshot 2022-10-26 at 14 48 16

Screenshot 2022-10-26 at 14 49 45

### CODE REVIEW

File: app/components/support_interface/reference_with_feedback_component.rb [hunk 1]
```diff
@@ -201,10 +192,11 @@ def feedback_row
     end
 
     def consent_row
+      return unless reference.feedback_provided?
       if feedback
         {
-          key: 'Given consent for research?',
-          value: consent_to_be_contacted_present,
```
Comment (tomas-stefano) [inline app/components/support_interface/reference_with_feedback_component.rb:207]: I think this method can be deleted now?

File: app/components/support_interface/reference_with_feedback_component.rb [hunk 2]
```diff
@@ -201,10 +192,11 @@ def feedback_row
...
+          key: 'Consent for research',
+          value: consent_to_be_contacted == true ? 'They can be contacted' : 'They have not given consent',
```
Comment (tomas-stefano) [inline app/components/support_interface/reference_with_feedback_component.rb:199]:
```
consent_to_be_contacted.present? ? 'They can be contacted' : 'They have not given consent'
```

File: app/components/support_interface/reference_with_feedback_component.rb [hunk 3]
```diff
@@ -156,9 +152,19 @@ def email_address_row
...
+        else
+          value += tag.p("This was confirmed by #{reference.name}", class: 'govuk-body')
+        end
+      end
+
       row = {
-        key: 'Relationship to candidate',
-        value: relationship,
+        key: 'How the candidate knows them and how long for',
+        value: value,
```
Comment (tomas-stefano) [inline app/components/support_interface/reference_with_feedback_component.rb:167]: Can we add tests for this method?

**Review by tomas-stefano — CHANGES_REQUESTED** [review]
LG just minor comments

File: app/components/support_interface/reference_with_feedback_component.rb (hunk 3 above)
Comment (frankieroberto) [reply app/components/support_interface/reference_with_feedback_component.rb:167]: Good idea. Done in [52c976a](https://github.com/DFE-Digital/apply-for-teacher-training/pull/7620/commits/52c976a1fc9e86daeca93e045b988edabb6f9bc6).

File: app/components/support_interface/reference_with_feedback_component.rb [hunk 4]
```diff
@@ -156,9 +152,19 @@ def email_address_row
...
     def relationship_row
+      value = tag.p(relationship, class: 'govuk-body')
+
+      if reference.feedback_provided?
+        value += if relationship_correction.present?
+                   tag.p("#{reference.name} said:", class: 'govuk-body') + tag.p(relationship_correction, class: 'govuk-body')
+                 else
+                   tag.p("This was confirmed by #{reference.name}", class: 'govuk-body')
+                 end
+      end
```
Comment (carlosmartinez) [inline app/components/support_interface/reference_with_feedback_component.rb:163]: suggest extracting a method to get this text, something along the lines of
```
      def relationship_text
        value = tag.p(relationship, class: 'govuk-body')
        return value unless reference.feedback_provided?

        if relationship_correction.present?
          value += tag.p("#{reference.name} said:", class: 'govuk-body')
          value += tag.p(relationship_correction, class: 'govuk-body')
        else
          value += tag.p("This was confirmed by #{reference.name}", class: 'govuk-body')
        end

        value
      end
```

Comment (frankieroberto) [reply app/components/support_interface/reference_with_feedback_component.rb]: Good idea. Done in [8a912d1](https://github.com/DFE-Digital/apply-for-teacher-training/pull/7620/commits/8a912d12dcb84fa471e766386210455a9acbcc2f), thanks. 👍

**Review by carlosmartinez — APPROVED** [review]

**Review by tomas-stefano — APPROVED** [review]

**Pull request merged**